### ISSUE TITLE
TypeName returns unexpected result with VBA.Collection

### ISSUE THREAD

**Issue by tdimitriou** [WaynePhillipsEA/twinbasic#84]
**Describe the bug**
The **_TypeName_** function returns **_Unknown_** instead of **_Collection_** for a VBA.Collection Object, although (TypeOf Obj Is VBA.Collection) returns True.

The result is also unexpected if the Variable is declared as **_Object_** (Dim Obj As **_Object_** = New Collection). In that case, the result of TypeName is **_Object_**. 

I have checked the TypeName function with several objects from external COM Libs and the result is as expected. I am not sure if the error resides in the TypeName function or the VBA.Collection Object itself.

**To Reproduce**
Code to reproduce the behavior (a reference to Microsoft Scripting Runtime is required):

```
Module modMain
    
    Sub Main()

        Dim Col As VBA.Collection = New VBA.Collection
        
        Debug.Print "TypeName(Col): " & TypeName(Col)
        Debug.Print "(TypeOf Col Is VBA.Collection): " & (TypeOf Col Is VBA.Collection)     
           
        Dim Col2 As Object = New VBA.Collection
        
        Debug.Print "TypeName(Col2): " & TypeName(Col2)
        Debug.Print "(TypeOf Col2 Is VBA.Collection): " & (TypeOf Col2 Is VBA.Collection)     

        Dim Dict As Scripting.Dictionary = New Scripting.Dictionary
        Debug.Print "TypeName(Dict): " & TypeName(Dict)
        Debug.Print "(TypeOf Dict Is Scripting.Dictionary): " & (TypeOf Dict Is Scripting.Dictionary)     
        
        Dim Dict2 As Object = New Scripting.Dictionary
        Debug.Print "TypeName(Dict2): " & TypeName(Dict2)
        Debug.Print "(TypeOf Dict2 Is Scripting.Dictionary): " & (TypeOf Dict2 Is Scripting.Dictionary)     
    End Sub
    
End Module
```

The output of the debug console is as follows:

```
TypeName(Col): Unknown
(TypeOf Col Is VBA.Collection): True
TypeName(Col2): Object
(TypeOf Col2 Is VBA.Collection): True
TypeName(Dict): Dictionary
(TypeOf Dict Is Scripting.Dictionary): True
TypeName(Dict2): Dictionary
(TypeOf Dict2 Is Scripting.Dictionary): True
```

**Expected behavior**
The function should always return the runtime type of the object. This should be the declared type, if any, or its Default Interface. 

**Desktop (please complete the following information):**
 - OS: Windows 10
 - twinBASIC compiler version 0.9.1882

**Comment by WaynePhillipsEA** [comment]
Thanks @tdimitriou, we'll get that fixed

**Comment by tdimitriou** [comment]
> Thanks @tdimitriou, we'll get that fixed

I am sure you will. Not even a single doubt about it!!! 👍👍👍

**Comment by tdimitriou** [comment]
@retailcoder That makes sense, and this is how it (a variable declared as object or variant) actually works, as far as I know.

If I recall correctly, In the cases where the object variable has been declared as **_Object_** or even **_Variant_**, the SomeVar.SomeMethod() call is using the IDispatch Interface to get the Ids of names (in the default interface) and then invoke the desired method.

The **_TypeName_** function, on the other hand, should return the runtime type name no matter if the Variable has been declared as Type, Object or even Variant. The runtime type should match the declared type, if explicitly declared, or the default interface if not.

BTW, @retailcoder I am a big fan of Rubberduck and you personally. Amazing job!

**Comment by WaynePhillipsEA** [comment]
Fixed in v0.9.1944.  Thanks for reporting.
